src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,49 @@
+@import '../utils/functions';
+@import '../utils/variables';
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: 750px;
+  padding: 10px;
+
+  $days: "mon" 0, "tue" 1, "wed" 2, "thu" 3, "fri" 4, "sat" 5, "sun" 6;
+
+  @each $day, $number-of-day in $days {
+    &--start-day--#{$day} .calendar__day:first-child {
+      margin-left: get-margin-size($number-of-day);
+    }
+  }
+
+  @for $length from 28 through 31 {
+    &--month-length--#{$length} .calendar__day:nth-child(n + #{$length + 1}) {
+      visibility: hidden;
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    box-sizing: border-box;
+    height: $square-size;
+    width: $square-size;
+    border: 1px solid black;
+    background-color: #eee;
+    list-style-type: none;
+    transition: background-color 0.5s, transform 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffc0cb;
+      transform: translateY(-20px);
+    }
+  }
+}
```

Comment:
Add [empty line in the end of file](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file) Fix all cases